### PR TITLE
Implemented skeleton code for csc class

### PR DESCRIPTION
**Pull request by hpleei**
1. Wrote a class skeleton for the CSC matrix
2. Defined methods/interfaces for the class: Constructor, print, getter
3. No destructor as it wouldn't need any dynamically allocated data structure for this object.
4. Using vector instead of plain array for more flexibilities and simplicity.

### CODE REVIEW

File: src/csc/cscsparsemat.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include <vector>
+#include <iostream>
+
+using namespace std;
+
+template<typename T>
+class CscSparseMatrix{
+    private:
+        vector<vector<T>> csc_matrix;
+    public:
```
Comment (Qicang) [inline src/csc/cscsparsemat.hpp:10]: The class should be the derived type of the Sparse matrix. Just the same as what Prof Kochunas @bkochuna  commented in #50. 
```
class CscSparseMatrix : public SparseMatrix 
```

**Review by Qicang — CHANGES_REQUESTED** [review]
Very nice work. But the class need to be modified to be the derived type of the base class ```SparseMatrix```

File: src/csc/cscsparsemat.hpp (hunk 1 above)
Comment (hpleei) [reply src/csc/cscsparsemat.hpp:10]: Got it. I just noticed that we are using some inheritance in this assignment. As professor's comment in #51 says, we would wait till his base class implementation gets done. And maybe @preethammohan9  can hop in working on that and I can help clarifying c++ inheritance concepts if he needs help.

**Review by bkochuna — APPROVED** [review]
Deferring to the review by @preethammohan9

**Comment by Qicang** [comment]
The checks are failed. Should fix this before merging into master @hpleei

**Comment by hpleei** [comment]
Found out that works after me has some problems and faced conflicts but finally fixed all of them and pushed. Was able to pass Travis CI -branch and Travis CI- Pull Request testing but don't know why both "c-i/travis-ci/pr" and "c-i/travis-ci/push" tests are pending and never gets tested (it keep saying it's being queued but it never goes to next step)

**Comment by preethammohan9** [comment]
I added back the csc subdirectory and tests to the build system and the tests again failed. Do we know what is going on?

**Review by Qicang — APPROVED** [review]

**Comment by Qicang** [comment]
@preethammohan9 All the checks passed, once you review all the changes, you can merge the code.

**Review by preethammohan9 — COMMENTED** [review]
As I myself edited a few files, I am happy it works now. Minor typo in test/cscTest.cpp line 6 "scopint" -> "scoping". Typo in comments in lines 15-16 of src/cscsparsemat.hpp. These have no consequence. Other than that, we have all the methods that are pure virtual in the parent class defined here. My one question @hpleei, @deanrp2 is, wouldn't we need the elements array, row idx array and col pointer array as attributes of the class, since we will be implementing the void assembleStorage() to "assemble" the matrix in some way using _buildCoeff?

**Review by preethammohan9 — APPROVED** [review]
We can defer the question in the review above to the implementation team, and any changes can be made in the future. In the interest of time, I am going to go ahead and merge this with main.

**Comment by preethammohan9** [comment]
Oops, I actually did a merge commit instead of a "Squash and Merge". Sincere apologies!!! :/

**Pull request merged**